weblate/addons/models.py

```diff
@@ -48,12 +48,20 @@
 
 
 class AddonQuerySet(models.QuerySet):
-    def filter_component(self, component):
+    def filter_component(self, component, project):
         return self.prefetch_related("event_set").filter(
-            (Q(component=component) & Q(project_scope=False))
-            | (Q(component__project=component.project) & Q(project_scope=True))
-            | (Q(component__linked_component=component) & Q(repo_scope=True))
-            | (Q(component=component.linked_component) & Q(repo_scope=True))
+            (Q(component=component) & Q(project=project) & Q(project_scope=False))
+            | (Q(project=project) & Q(project_scope=True))
+            | (
+                Q(component__linked_component=component)
+                & Q(repo_scope=True)
+                & Q(project=project)
+            )
+            | (
+                Q(component=component.linked_component)
+                & Q(repo_scope=True)
+                & Q(project=project)
+            )
```

Comment:
The final query here should be:

```suggestion
            Q(component=component)
            | Q(project=project)
            | (Q(component__linked_component=component) & Q(repo_scope=True))
            | (Q(component=component.linked_component) & Q(repo_scope=True))
```

This needs dropping `project_scope` and using `project` properly instead.

* `component` would be NULL for project scoped add-ons
* `project` would be NULL for component scoped add-ons